src/libraries/Directory.Build.targets

```diff
@@ -233,4 +233,23 @@
     <None Include="$(MSBuildProjectDirectory)\**\*.cs"
           Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Compile)" />
   </ItemGroup>
+
+  <PropertyGroup>
+    <SkipLocalsInit Condition="'$(IsSourceProject)' == 'true' and '$(MSBuildProjectExtension)' == '.csproj' and '$(IsNETCoreApp)' == 'true' and '$(TargetFramework)' == '$(NetCoreAppCurrent)'">true</SkipLocalsInit>
+  </PropertyGroup>
+
+  <!--Instructs compiler not to emit .locals init, using SkipLocalsInitAttribute.-->
+  <Choose>
+    <When Condition="'$(SkipLocalsInit)' == 'true'">
+      <PropertyGroup >
+        <!-- This is needed to use the SkipLocalsInitAttribute. -->
+        <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+      </PropertyGroup>
+
+      <ItemGroup>
+        <Compile Include="$(CommonPath)ShouldSkipLocalsInit.cs" Link="Common\ShouldSkipLocalsInit.cs" />
```

Comment:
The only last NIT I have is that I would probably name this file `SkipLocalsInit.cs` instead in order to match the property condition and the attribute name but feel free to use your best judgement.

Reply:
> name this file SkipLocalsInit.cs 

+1